src/app/server/Server.cpp

```diff
@@ -95,30 +93,12 @@ class DeviceTypeResolver : public chip::Access::AccessControl::DeviceTypeResolve
         {
             if (type->deviceTypeId == deviceType)
```

Comment:
This looks like the second copy of this function we have sitting around somewhere.  Should this just be a non-virtual member method on DataModel::Provider?  Followup for that, obviously.

Reply:
Started #36327 to track these things.

Overall I would be weary of adding too much into DataModel::Provider if at all possible, however all for code reuse however we figure out it should look like.